src/main/scala/beam/sim/vehiclesharing/AvailabilityBasedRepositioning.scala

```diff
@@ -0,0 +1,109 @@
+package beam.sim.vehiclesharing
+import beam.agentsim.agents.vehicles.{BeamVehicle, BeamVehicleType}
+import beam.agentsim.events.SpaceTime
+import beam.agentsim.infrastructure.TAZTreeMap.TAZ
+import beam.router.BeamSkimmer
+import beam.router.Modes.BeamMode
+import beam.sim.BeamServices
+import org.matsim.api.core.v01.Id
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+private[vehiclesharing] class AvailabilityBasedRepositioning(
+  beamSkimmer: BeamSkimmer,
+  beamServices: BeamServices
+) extends RepositionAlgorithm {
+
+  case class RepositioningRequest(taz: TAZ, availableVehicles: Int, shortage: Int)
+
+  val LABEL: String = "availability"
+
+  override def getVehiclesForReposition(
+    startTime: Int,
+    endTime: Int,
+    repositionManager: RepositionManager
+  ): List[(BeamVehicle, SpaceTime, Id[TAZ], SpaceTime, Id[TAZ])] = {
+    val relocate = beamServices.iterationNumber > 0 || beamServices.beamConfig.beam.warmStart.enabled
+    val oversuppliedTAZ =
+      mutable.TreeSet.empty[RepositioningRequest](Ordering.by[RepositioningRequest, Int](_.availableVehicles))
+    val undersuppliedTAZ =
+      mutable.TreeSet.empty[RepositioningRequest](Ordering.by[RepositioningRequest, Int](_.shortage))
+
+    beamServices.tazTreeMap.getTAZs.foreach { taz =>
+      if (relocate) {
```

Comment:
Nothing happens in the loop if `relocate` is false, so it can be taken out of loop

Reply:
okay I changed this code too